tools/symbols_map_generator.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python3
+
+# Copyright © Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or 3
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import glob
+import os
+import argparse
+from pathlib import Path
+from typing import Literal, TypedDict, get_args
+import clang.cindex
+
+_logger = logging.getLogger(__name__)
+
+LibraryName = Literal[
+    "miral", "mir_common", "mir_core", "miroil", "mir_platform",
+    "mir_renderer", "mir_renderers", "mir_server", "mir_test", "mir_wayland"]
+
+
+class LibraryInfo(TypedDict):
+    external_headers: list[str]
+    internal_headers: list[str]
+    external_output_file: str
+    internal_output_file: str
+
+
+# Directory paths are relative to the root of the Mir project
+library_to_data_map: dict[LibraryName, LibraryInfo] = {
+    "miral": {
+        "external_headers": [
+            "include/miral"
+        ],
+        "internal_headers": [
+            "src/miral"
+        ],
+        "external_output_file": "src/miral/symbols_external.map",
+        "internal_output_file": "src/miral/symbols_internal.map"
+    }
+}
+
+
+def get_absolute_path_form_project_path(project_path: str) -> str:
+    script_path = os.path.dirname(os.path.realpath(__file__))
+    path = Path(script_path).parent.joinpath(project_path)
+    return path.absolute().as_posix()
```

Comment:
*from

But also, stick to `Path`:
```python
return Path(__file__).parent / project_path
```

You can then [`.glob()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) that, no need for the `glob` module.

Reply:
It could be an arg if you want? Although it's not like it will be ran from anywhere else IMO